tools/aapt_lite/src/main/java/com/grab/aapt/main.kt

```diff
@@ -37,14 +38,13 @@ enum class Tool {
 }
 
 fun main(args: Array<String>) {
-    Worker.from(args = args.toList()).run { cliArgs ->
-        try {
-            Tool.valueOf(cliArgs.first())
-                .call(cliArgs.drop(1).toTypedArray())
-            Status.Success
-        } catch (e: Exception) {
-            e.printStackTrace()
-            Status.Failure
-        }
+    Worker.create(args,extractAction()).run()
+}
+
+fun extractAction(): (args: Array<String>) -> Unit {
+    return fun (args: Array<String>) {
+        val cliArgs = commandLindArgs(args.filter { it != PERSISTENT_WORKER }.toTypedArray())
```

Comment:
Can we do this inside `tools/worker` and ensure the args received in callback already has `--persistent_worker` filtered?

Otherwise we would need to write this for every worker tool we write. The previous implementation did this for example. 

https://github.com/grab/grab-bazel-common/blob/8d7270658a9e7a63a2cb5115e1162a6905240ddc/tools/worker/src/main/kotlin/io/bazel/Worker.kt#L48